core/utils_test.py

```diff
@@ -794,3 +794,85 @@ def test_url_open(self) -> None:
         self.assertEqual(response.getcode(), 200) # type: ignore[attr-defined]
         self.assertEqual(
             response.url, 'http://www.google.com') # type: ignore[attr-defined]
+
+    def test_get_random_int(self) -> None:
+        self.assertLess(utils.get_random_int(5), 5)
+        self.assertGreaterEqual(utils.get_random_int(5), 0)
+        self.assertLess(utils.get_random_int(True), 1)
+        self.assertGreaterEqual(utils.get_random_int(True), 0)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only positive integers allowed'):
+            utils.get_random_int(-1)
+
+    def test_get_random_choice(self) -> None:
+        list_instance = [1, 5, 9, 11, 15]
+        list_instance2: List[str] = []
+        self.assertIn(utils.get_random_choice(list_instance), list_instance)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only non-empty lists allowed'):
+            utils.get_random_choice(list_instance2)
+
+    def test_get_human_readable_time_string(self) -> None:
+        """Should we test for negative time."""
+        self.assertEqual('December 12 06:42:12', utils.get_human_readable_time_string(944980932342.38))  # pylint: disable=line-too-long
+
+    def test_generate_new_session_id(self) -> None:
+        self.assertEqual(24, len(utils.generate_new_session_id()))
+        self.assertEqual(type(utils.generate_new_session_id()).__name__, 'str')
+
+    def test_require_valid_name_with_incorrect_input(self) -> None:
+        """ADD TEST ESCAPE SEQUENCE CHARACTERS CHECK."""
```

Comment:
This looks like another TODO comment

Reply:
No, actually it is possible to have characters like '\\\b' or '\\\u0001' in the name; when such characters are present we should throw an error. However, when I tried testing it the exception wasn't working properly. 
For example the following test:
![escape sequence -(1)](https://user-images.githubusercontent.com/81075125/159548073-2985db1e-fb82-48e3-8b3d-babca757f2a9.JPG)

Gives the following error:
![escape sequence -(2)](https://user-images.githubusercontent.com/81075125/159548091-66d70c48-0a7b-411d-9cba-4d91b8899d0f.JPG)

Reply:
This looks like important functionality to check, so I think we need to write a test for this behavior. The end of the error message is cut off in your screenshot, but from what I can see the errors seem to match, so I don't know why they failed. We have a wiki on debugging backend tests that might help you investigate: https://github.com/oppia/oppia/wiki/Debug-backend-tests

Reply:
Done